jerry-core/ecma/builtin-objects/ecma-builtin-date-prototype.cpp

```diff
@@ -38,6 +42,55 @@
  */
 
 /**
+ * Insert leading zeros to a string of a number if needed.
+ */
+static void
+ecma_date_insert_leading_zeros (ecma_string_t **str_p, /**< input/output string */
+                                ecma_number_t num, /**< input number */
+                                int32_t div) /**< "length" {10, 100, 1000} */
+{
+  JERRY_ASSERT (div == 10 || div == 100 || div == 1000);
+
+  while (num < div && div > 1)
+  {
+    div /= 10;
+
+    ecma_string_t *zero_str_p = ecma_new_ecma_string_from_number (ECMA_NUMBER_ZERO);
```

Comment:
ecma_new_ecma_string_from_uint32 would be better in the case, as no additional pools chunk would be allocated for floating point value.